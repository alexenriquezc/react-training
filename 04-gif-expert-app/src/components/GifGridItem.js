import React from 'react'

export const GifGridItem = ({ title, url }) => {
    const rand = Math.random();
    return (
        <div className="card animate__animated animate__fadein" key={rand}>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
