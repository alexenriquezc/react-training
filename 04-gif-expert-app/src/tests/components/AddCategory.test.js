import { shallow } from 'enzyme';
import React from 'react';
import {AddCategory} from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Testing in <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach( ()=> {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    });
    
    test('should shows correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('input text shoud changes', () => {
        const textInput = wrapper.find('input');
        const value = "Hola Mundo";
        textInput.simulate('change', { target: { value } } );

        expect(wrapper.find('p').text().trim()).toBe(value);
    });        
        
    test("shouldn't post info", () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });
    
})
    