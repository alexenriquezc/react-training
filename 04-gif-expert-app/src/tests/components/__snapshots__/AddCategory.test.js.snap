// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <AddCategory/> should shows correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
